app/models/school.rb

```diff
@@ -9,7 +9,19 @@ class School < ApplicationRecord
             presence: true,
             uniqueness: true
 
+  # Scopes
+  scope :search, ->(q) { includes(:gias_school).references(:gias_schools).where("gias_schools.search @@ websearch_to_tsquery('unaccented', ?)", q) }
+
   # Instance Methods
+  delegate :address_line1, :address_line2, :address_line3, to: :gias_school
+  delegate :administrative_district_name, :closed_on, :establishment_number, to: :gias_school
+  delegate :funding_eligibility, :induction_eligibility, :in_england, to: :gias_school
+  delegate :local_authority_code, :local_authority_name, to: :gias_school
+  delegate :name, :primary_contact_email, to: :gias_school
+  delegate :opened_on, :phase_name, :postcode, :type_name, to: :gias_school
+  delegate :secondary_contact_email, :section_41_approved, :status, to: :gias_school
+  delegate :ukprn, :website, to: :gias_school
```

Comment:
Could we use the GIAS School for the school data and not `delegate`?
Personally I don't like delegate in database models as I feel it obsfucates relationships and the sources of data and you can't use them in queries. It would be better to use a presenter type of pattern when really necessary and keep the models clean and simple IMO.

Reply:
Actually I have the opposite view in the School model.

We almost always work with School in the service instead of GIAS::Schools. 
If a School instance is not able to respond to messages about their basic properties (name, status, urn... etc), what is the purpose of having the School class?

On the other hand, delegation allows encapsulation of a School entity so the rest of the service doesn't need to know about the internals on how their properties are stores, where, etc, preventing things like school.gias_school.name in favour of school.name according to the Law of Demeter.

Reply:
I'm normally against delegation (and always against it when [prefixes are involved](https://www.rubydoc.info/docs/rails/Module:delegate)) but here I think it's ok as the school model is a kind of a empty 'pointer' at the true GIAS school.

I think the above snippet could be neater with a single list alphabetically sorted delegations though:

```suggestion
  delegate
    :address_line1,
    :address_line2,
    :address_line3
    :administrative_district_name,
    :closed_on,
    :establishment_number,
    :etc,
    to: :gias_school
```

Reply:
True this is an unusual case, but I'd like us to avoid it in DB models as a general rule.